sds/src/secondary_validation/jwt_expiration_checker.rs

```diff
@@ -0,0 +1,138 @@
+use crate::secondary_validation::Validator;
+use base64::{engine::general_purpose::URL_SAFE_NO_PAD, Engine};
+use chrono::Utc;
+use serde_json::Value as JsonValue;
+
+pub struct JwtExpirationChecker;
+
+const SEGMENTS_COUNT: usize = 3;
+
+impl Validator for JwtExpirationChecker {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        if let Some((_, payload)) = decode_segments(regex_match) {
+            let expiration = payload.get("exp");
+            if let Some(exp) = expiration {
+                if exp.is_i64() {
+                    let exp_int: i64 = exp.as_i64().unwrap();
+                    let now = Utc::now().timestamp();
+                    exp_int > now
+                } else {
+                    // if the expiration claim is not an integer, we consider it as an invalid match
+                    false
+                }
```

Comment:
You can drop the unwrap completely by relying on the option stat, inline the `exp` variable, and use the same name in the for loop, it's quite comment to call the unwrapped option with the same name as the option (`exp` in the suggestion). 

```suggestion
            if let Some(exp) = payload.get("exp") {
                if let Some(exp) = exp.as_i64() {
                    let now = Utc::now().timestamp();
                    exp > now
                } else {
                    // if the expiration claim is not an integer, we consider it as an invalid match
                    // exp is a reserved claim for NumericDate (https://www.rfc-editor.org/rfc/rfc7519#section-4.1.4)
                    // The NumericDate is the UNIX timestamp (https://www.rfc-editor.org/rfc/rfc7519#section-2)
                    false
                }
```